src/controllers/routine/update-lecture.controller.ts

```diff
@@ -0,0 +1,82 @@
+import { PrismaClient } from '@prisma/client';
+
+import Lecture from '../../models/lecture.model';
+import asyncHandler from '../../utils/asyncHandler';
+import throwError from '../../utils/throw-error';
+
+const { lecture } = new PrismaClient();
+
+const days = [
+  'monday',
+  'tuesday',
+  'wednesday',
+  'thursday',
+  'friday',
+  'saturday',
+  'sunday',
+];
+
+/**
+ * @auth required
+ * @route {POST} /lecture/update_lecture_details
+ * @body {subject: string, day: string, teacherId: string, startTime: date, endTime: date}
+ * @returns {updated count}
+ */
+
+const updateLectureDetails = asyncHandler(async (req, res) => {
+  try {
+    const { id, subject, day, teacherId, startTime, endTime } =
+      req.body as Lecture;
+    const lectureDetails = {} as Partial<Lecture>;
+
+    const isPresent = await lecture.findFirst({ where: { id } });
+
+    if (!isPresent) {
+      res.status(404);
+      return throwError('Lecture not found');
+    }
+
+    if (subject) {
+      lectureDetails.subject = subject;
+    }
+    if (day && days.includes(day.toLowerCase())) {
```

Comment:
why double validation and 
check in one place api routes or in controller

Reply:
taking a look